import-scripts/annotation_suite/merge_mafs.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+import argparse
+# Script for mergining MAFs
+# Creates one aggregate header taken from column names acorss MAFs
+# Writes out rows while mapping to header
+# NAs and equivalents replaced with ''
+# - sequenced samples header???
+TUMOR_SAMPLE_BARCODE_COLUMN = "Tumor_Sample_Barcode"
+HUGO_SYMBOL_COLUMN = "Hugo_Symbol"
+ENTREZ_GENE_ID_COLUMN = "Entrez_Gene_Id"
+
+def process_datum(val):
+    """ Cleans up datum. """
+    try:
+        vfixed = val.strip()
+    except AttributeError:
+        vfixed = ""
+    if vfixed in ["", "NA", "N/A", None]:
+        return ""
+    return vfixed
+
+def get_file_header(data_file):
+    """
+        Returns header from MAF
+        Assumes file is tab-delimited
+        Assumes header is first non-commented line
+    """
+    header = []
+    with open(data_file, "r") as header_source:
+        for line in header_source:
+            if not line.startswith("#"):
+                header = list(map(str.strip, line.rstrip().split("\t")))
+                break
+    return header
+
+def merge_headers(data_filenames):
+    """
+        Generates a merged header from all input files.
+        * Also ensures that "Hugo_Symbol" and "Entrez_Gene_Id"
+          are the first 2 columns in the merged MAF header.
+    """
+    merged_header = [HUGO_SYMBOL_COLUMN, ENTREZ_GENE_ID_COLUMN]
+    for fname in data_filenames:
+        columns_to_add = [column for column in get_file_header(fname) if column not in merged_header]
+        merged_header.extend(columns_to_add)
+    return merged_header
+
+def merge_input_mafs(input_mafs, output_maf_filename):
+    """ Generates a merged MAF given a list of input MAFs. """
+    merged_header = merge_headers(input_mafs)
+    sequenced_samples = set()
+
+    rows_to_write = []
+    for input_maf in input_mafs:
+        print('Loading data from ' + input_maf)
+        with open(input_maf, "r") as maf:
```

Comment:
It feels wrong (maybe it is fine) to open the file here, keep an open file handle to the file and then call get_file_header() below ... which will **also** open the file and have an open file handle. In other words, this script will have two simultaneous open file handles to the same file. I wonder if some operating systems will prevent that from working correctly. They are both "read" mode handles, which **should** not interfere with each other in theory. But maybe it would be reasonable to move the call to get_file_header above this line?

Reply:
I'm following the same pattern we've used in our other scripts